ultimatepython/classes/encapsulation.py

```diff
@@ -0,0 +1,117 @@
+"""    
+    Encapsulation is a feature of OOP that allows you to hide the
+    implementation details of a class from its users.
+    Encapsulation allows us to limit the access of certain attributes 
+    within a class. This prevents users from directly accessing and modifying such
+    attributes from outside the class. Instead, users must use methods to access and
+    modify attributes.
+"""
+import random
+
+class BankAccount:
+    def __init__(self, account_holder_name: str):
+        """
+            In python, a class attribute can be made private by prefixing it with two underscores.
+            This makes it inaccessible to users outside the class.
+            By default, class variables are public. Therefore, they can be accessed and modified
```

Comment:
Make sure to be consistent when referring to class attributes vs. variables

Reply:
Also please don't indent the start of comments by 4 spaces, keep the triple quote in line with the rest of paragraph.